skbio/metadata/_interval.py

```diff
@@ -456,7 +467,9 @@ class IntervalMetadata():
     Interval(interval_metadata=..., locations=[(3, 9)], \
 boundaries=[(True, True)], metadata={'gene': 'sagB'})
     """
-    def __init__(self):
+    def __init__(self, upper_bound):
+        self.upper_bound = upper_bound
```

Comment:
Can you make this a private attribute `self._upper_bound`, and expose it as a public, read-only property? That way users can't change it after construction (e.g. to a smaller upper bound).